src/main/java/dal/fatih/todoly/HibernateTaskRepository.java

```diff
@@ -0,0 +1,112 @@
+package dal.fatih.todoly;
+
+
+import javax.persistence.*;
+import javax.persistence.criteria.CriteriaBuilder;
+import javax.persistence.criteria.CriteriaQuery;
+import javax.persistence.criteria.Root;
+import javax.persistence.metamodel.EntityType;
+import java.sql.Date;
+import java.time.Instant;
+import java.util.List;
+import java.util.Locale;
+
+public class HibernateTaskRepository implements TaskRepository {
+
+    private final EntityManagerFactory entityManagerFactory
+            = Persistence.createEntityManagerFactory("Todoly");
+    private EntityManager entityManager = entityManagerFactory.createEntityManager();
+    private CriteriaBuilder cb;
+    private CriteriaQuery<Task> cq;
+    private Root<Task> taskRoot;
+
+
+    public void getBaseQuery() {
+        cb = entityManager.getCriteriaBuilder();

```

Comment:
Never mutate objects. Create them only at once. Next time it should be created from scratch. 

```suggestion
        CriteriaBuilder cb = entityManager.getCriteriaBuilder();
```